dag-cbor/src/lib.rs

```diff
@@ -36,6 +36,11 @@ pub trait DagCbor: Encode<DagCborCodec> + Decode<DagCborCodec> {}
 
 impl<T: Encode<DagCborCodec> + Decode<DagCborCodec>> DagCbor for T {}
 
+/// Bytes is sequence of byte values.
+///
+/// Implements Encode and Decode to/from CBOR byte strings
+pub type Bytes = Box<[u8]>;
```

Comment:
I'm a little confused as to why this is needed. It seems like Vec<u8> is covered. Also why Box<[u8]> and not Vec<u8>.

Reply:
It stems from here https://github.com/ipld/libipld/blob/master/dag-cbor/src/decode.rs#L297 and here https://github.com/ipld/libipld/blob/master/dag-cbor/src/decode.rs#L321

Using a `Vec<u8>` creates an IPLD list of u8 values, where using `Box<[u8]>` creates an IPLD byte sequence. 

We could change the cbor From implementations but this was the smallest change I could make to existing code.

Reply:
LIttle odd that they didn't handle Vec<u8> as a byte string, but is there a reason we need a ByteString here, and an array of u8 isn't sufficient. A large number of libraries work on Vec<u8> or bytes::Bytes, so to use this, they'll have to be regularly calling into_boxed_slice() to work with the types specified here. 

dag-pb actually uses bytes::Bytes. Oddly, it also returns a Box<[u8]> for the payload here https://github.com/ipld/libipld/blob/master/dag-pb/src/codec.rs#L78 rather than the Vec or a Bytes structure.